src/Microsoft.ML/Models/TrainTestEvaluator.cs

```diff
@@ -0,0 +1,181 @@
+
+                        Var<IPredictorModel> predictorModel;
+                        if (transformModels.Count != 0)
+                        {
+                            var localModelInput = new Transforms.ManyHeterogeneousModelCombiner
+                            {
+                                PredictorModel = predictorDataStep.Model,
+                                TransformModels = new ArrayVar<ITransformModel>(transformModels.ToArray())
+                            };
+                            var localModelOutput = subGraph.Add(localModelInput);
+                            predictorModel = localModelOutput.PredictorModel;
+                        }
+                        else
+                            predictorModel = predictorDataStep.Model;
+                        firstModel = predictorModel;
+
+                        var scorer = new Transforms.Scorer
+                        {
+                            PredictorModel = predictorModel
+                        };
+
+                        var scorerOutput = subGraph.Add(scorer);
+                        lastTransformModel = scorerOutput.ScoringTransform;
+                        step = new ScorerPipelineStep(scorerOutput.ScoredData, scorerOutput.ScoringTransform);
+                        transformModels.Clear();
+                    }
+                }
+
+                if (transformModels.Count > 0)
+                {
+                    if (lastTransformModel != null)
+                        transformModels.Insert(0, lastTransformModel);
+
+                    var modelInput = new Transforms.ModelCombiner
+                    {
+                        Models = new ArrayVar<ITransformModel>(transformModels.ToArray())
+                    };
+
+                    var modelOutput = subGraph.Add(modelInput);
+                    lastTransformModel = modelOutput.OutputModel;
+                }
+
+                var experiment = environment.CreateExperiment();
+                
+                TrainingData = (loaders[0].ApplyStep(null, experiment) as ILearningPipelineDataStep).Data;
+                TestingData = (testData.ApplyStep(null, experiment) as ILearningPipelineDataStep).Data;
+                Nodes = subGraph;
+                TransformModel = null;
+                Inputs.Data = firstTransform.GetInputData();
+                Outputs.TransformModel = lastTransformModel;
+                Outputs.UseTransformModel = true;
+                var crossValidateOutput = experiment.Add(this);
+                experiment.Compile();
+                foreach (ILearningPipelineLoader loader in loaders)
+                {
+                    loader.SetInput(environment, experiment);
+                }
+                testData.SetInput(environment, experiment);
+
+                experiment.Run();
+
+                TrainTestEvaluatorOutput<TInput, TOutput> tteo = new TrainTestEvaluatorOutput<TInput, TOutput>();
```

Comment:
>TrainTestEvaluatorOutput<TInput, TOutput> tteo  [](start = 16, length = 47)

var result or var output or var evalOutput.

Reply:
i would still prefer var.

---
In reply to: [191005451](https://github.com/dotnet/machinelearning/pull/212#discussion_r191005451) [](ancestors = 191005451)